packages/vsce/src/utils/profileManagement.ts

```diff
@@ -69,185 +66,158 @@ export class ProfileManagement {
-                }
-              }
+        window.showErrorMessage(`Cannot find plex ${profile.profile.cicsPlex} for profile ${profile.name}`);
+
+        throw new Error("Plex Not Found");
+      }
+    } else if (profile.profile.regionName) {
+      /**
+       * Region but no plex - Single region system, use that
+       */
+      const singleRegionJson = await getResource(session, {
+        resource: "CICSRegion",
+        regionName: profile.profile.regionName,
+      })
+      if (singleRegionJson.response.records?.cicsregion?._attributes != null) {
+        infoLoaded.push({
+          plexname: null,
+          regions: [singleRegionJson.response.records?.cicsregion?._attributes],
+          group: false,
+        });
+      } else {
+        window.showErrorMessage(`Cannot find region ${profile.profile.regionName} for profile ${profile.name}`);
+
+        throw new Error("Region Not Found");
+      }
+    } else {
+      /**
+       * Nothing given - Test if plex and find all info
+       */
+      try {
+        // Plex
+        const cicsplexJson = await getResource(session, {
+          resource: "CICSCICSPlex",
+        });
+        if (cicsplexJson.response.records?.cicscicsplex != null) {
+          const returnedPlexes = cicsplexJson.response.records?.cicscicsplex.map((item: { _attributes: any }) => item._attributes);
+          const uniqueReturnedPlexes = returnedPlexes.filter(
+            (plex: any, index: number) => index === returnedPlexes.findIndex((found: any) => found.plexname === plex.plexname)
+          );
+          for (const plex of uniqueReturnedPlexes) {
+            try {
+              // Regions are empty because we only load Plex when session is expanded
+              infoLoaded.push({
+                plexname: plex.plexname,
+                regions: [],
+                group: false,
+              });
+            } catch (error) {
+              console.log(error);
             }
-          } else {
-            // Not Plex
-            const singleRegion = await ProfileManagement.makeRequest(`/CICSRegion`, config);
-            const jsonFromXml = ProfileManagement.cmciResponseXml2Json(singleRegion.data);
-            const returnedRegion = jsonFromXml.response.records.cicsregion._attributes;
-            infoLoaded.push({
-              plexname: null,
-              regions: [returnedRegion],
-              group: false,
-            });
           }
-        } catch (error) {
-          // Not Plex - Could be error
-          try {
-            const singleRegion = await ProfileManagement.makeRequest(`/CICSRegion`, config);
-            const jsonFromXml = JSON.parse(xml2json(singleRegion.data, { compact: true, spaces: 4 }));
-            if (!jsonFromXml) {
-              throw error;
-            }
-            const returnedRegion = jsonFromXml.response.records.cicsregion._attributes;
+        } else {
+          // Not Plex
+          const singleRegionJson = await getResource(session, {
```

Comment:
The implementation of `getResource` at this version requires a region name causing failures.
```
ImperativeExpect.toBeDefinedAndNonBlank(parms.regionName, "CICS Region name", "CICS region name is required");
```

Being unfamiliar with maintaining this V2 version, what's the best approach here? The reason we initially adapted getResource in V3 was to accommodate not specifying a region name 🤔

Reply:
Ahhh the second PR adds these changes in..! Perhaps that's necessary to complete the removal of Axios

Reply:
Hey @AndrewTwydell,

This PR is meant to just remove axios without introducing enhancements. 

The other PR (#275) should not be merged as it contains enhancements for the V2 (maintenance) branch 😅 
I opened it just as an example, but would love to close it in favor of this one 😅